### DISCUSSION TITLE
Discord Rich Presence not working as expected

### DISCUSSION THREAD

**Discussion by 0xdeadbeer** [ppy/osu#16277] [Q&A] [answered]
After playing a while with osu! lazer (latest commit) I noticed that on Discord no matter if I'm playing a song or not, my osu! status will always appear as "Idle". And yes I already set "full" on "Discord Rich Presence" option which is located inside osu!'s settings. I'll take a look at the code later and see if I can fix the problem on my own or report anything back.. so far that's it
![image](https://user-images.githubusercontent.com/64986162/147658160-c79d45a3-9238-41cf-971c-11874c28813d.png)

**Comment by peppy** [comment, marked as answer]
Are you logged in in lazer? Can please attach your log files after reproducing the issue (especially `network.log`)?

**Reply by 0xdeadbeer** [reply]
So.. yeah I've done some searching inside the code and apparently I gotta be signed in to make Discord Rich Presence on osu! work? I'm not sure if I'm right but that's at least what I understood from the code I read. 


( notice that the updater.log && database.log files didn't get any logs from osu! so there was no point in appending them ) 
( also.. I removed the logs inside runtime.log for which keys I pressed as I was logging in multiple times )
[performance.log](https://github.com/ppy/osu/files/7792416/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7792417/runtime.log)
[network.log](https://github.com/ppy/osu/files/7792418/network.log)

**Reply by peppy** [reply]
Yes, right now you do need to be logged in. Can you confirm that it works after logging in?

**Reply by 0xdeadbeer** [reply]
Strangely enough I can't login into my account.. I tried several times retyping my password, changing it, etc.. but neither of those did work on osu! lazer (I can login into my osu account on the official website though)
On top of that if I fail my sign in (inside osu! lazer) a number times it does say that my ip has been blocked so either I'm the one that can't type credentials correctly or there's another problem inside the code.. I'll report back if I get any luck with logging in..

**Reply by 0xdeadbeer** [reply]
So... a thing that I noticed is that when I type my password and then look at the `runtime.log` file it recognizes the Y that I pressed as Z.. I'm not sure if that's the problem as in the game when ever I click Y a Y is shown to the textbox... 
I use a Slovenian keyboard layout btw (windows 10)

**Reply by nahkd123** [reply]
> Strangely enough I can't login into my account..

If you're building from source, make sure to use "Release" configuration. After that you should be able to login with custom build.

**Reply by 0xdeadbeer** [reply]
> > Strangely enough I can't login into my account..
> 
> If you're building from source, make sure to use "Release" configuration. After that you should be able to login with custom build.

Sorry but how do I do that?

**Reply by peppy** [reply]
Use a pre-built release, or create an account on the development server.

**Reply by 0xdeadbeer** [reply]
> Use a pre-built release, or create an account on the development server.

What's the point of using the pre-built release if I can't see it's source code, either way forget about it, it's not that important for me to log in. 
Also forgot to ask the question.. why didn't we add the option to always show the discord status no matter if the user is signed in or not?
It doesn't make sense in my opinion to leave it on idle no matter if the person is playing or not, can you explain why left it like that?

**Reply by peppy** [reply]
The reasoning for this is that it blocks status when the user is set to "offline", which currently covers the case of both "logged out" and "hiding online status".

If it's a common thing that users want to broadcast their game status on discord without being logged in, we could consider changing this. Though it is very rare for a user to be playing osu! without being logged in, as it means you aren't tracking any statistics nor able to save scores.

**Reply by 0xdeadbeer** [reply]
> The reasoning for this is that it blocks status when the user is set to "offline", which currently covers the case of both "logged out" and "hiding online status".
> 
> If it's a common thing that users want to broadcast their game status on discord without being logged in, we could consider changing this. Though it is very rare for a user to be playing osu! without being logged in, as it means you aren't tracking any statistics nor able to save scores.

Oh okay got that! thanks for answering!

**Reply by shrublet** [reply]
> The reasoning for this is that it blocks status when the user is set to "offline", which currently covers the case of both "logged out" and "hiding online status".
> 
> If it's a common thing that users want to broadcast their game status on discord without being logged in, we could consider changing this. Though it is very rare for a user to be playing osu! without being logged in, as it means you aren't tracking any statistics nor able to save scores.

sorry for the necro, but was also curious about offline play and discord rpc. iirc stable still displays full discord rpc when playing offline. in this situation, it would be nice to have parity with stable. if you're offline and want to hide what you're doing, it still shows you're playing osu, so i feel people would just opt to remove game activity directly in discord. thanks for your consideration.
